api_test.go

```diff
@@ -37,6 +27,12 @@ func WithServer(t *testing.T, f func(string)) {
 		time.Sleep(50 * time.Millisecond)
 	}
 
+	// Need to assign here since we need to call NewProxy to set the client.
+	testProxy.Name = "mysql_master"
+	testProxy.Listen = "localhost:3310"
+	testProxy.Upstream = "localhost:20001"
+	testProxy.Enabled = true
```

Comment:
This kind of sucks, but I'm not sure how to create a better interface unless we require passing a proxy object to `NewProxy`, which is also weird...